server.go

```diff
@@ -817,10 +817,10 @@ func (s *Server) Serve(lis net.Listener) error {
 		// Make sure we account for the goroutine so GracefulStop doesn't nil out
 		// s.conns before this conn can be added.
 		s.serveWG.Add(1)
-		go func() {
+		go func(rawConn net.Conn) {
```

Comment:
I don't think there's any risk. `rawConn` is a local var defined in the loop body.